File: src/fr/kwizzy/webserver/CloseCompletionHandler.scala
package fr.kwizzy.webserver

import java.nio.channels.{AsynchronousSocketChannel, CompletionHandler}

/**
  * Par Alexis le 19/12/2016.
  */
object CloseCompletionHandler extends CompletionHandler[Integer, AsynchronousSocketChannel] {

  override def completed(result: Integer, attachment: AsynchronousSocketChannel): Unit =
  {
    try {
      attachment.close()
    }
    catch {
      case e : Exception => failed(e, attachment)
    }
  }

  override def failed(exc: Throwable, attachment: AsynchronousSocketChannel): Unit =
  {
    exc.printStackTrace()
  }
}

File: src/fr/kwizzy/japache/CloseCompletionHandler.java
package fr.kwizzy.japache;

import java.io.IOException;
import java.nio.channels.AsynchronousSocketChannel;
import java.nio.channels.CompletionHandler;

/**
 * Par Alexis le 18/12/2016.
 */

public class CloseCompletionHandler implements CompletionHandler<Integer, AsynchronousSocketChannel>
{
    public static final CloseCompletionHandler INSTANCE = new CloseCompletionHandler();

    private CloseCompletionHandler()
    {}


    @Override
    public void completed(Integer result, AsynchronousSocketChannel attachment)
    {
        try {
            attachment.close();
        } catch (IOException e) {
            failed(e , attachment);
        }
    }

    @Override
    public void failed(Throwable exc, AsynchronousSocketChannel attachment)
    {
        exc.printStackTrace();
    }
}
